src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -92,18 +92,37 @@ private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
             bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
             string exeName = isWindows ? "dotnet.exe" : "dotnet";
 
-            // We will generally find the dotnet exe on the path, but on linux, it is often just a 'dotnet' symlink (possibly even to more symlinks) that we have to resolve
-            // to the real dotnet executable.
-            // This will work as often as just invoking dotnet from the command line, but we can be more confident in finding a dotnet executable by following
-            // https://github.com/dotnet/designs/blob/main/accepted/2021/install-location-per-architecture.md
-            // This can be done using the nethost library. We didn't do this previously, so I did not implement this extension.
-            foreach (string dir in Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator))
-            {
-                string filePath = Path.Combine(dir, exeName);
-                if (File.Exists(Path.Combine(dir, exeName)))
+            // First check for the DOTNET_ROOT environment variable, as it's often there as with, for example, dotnet format.
+            string dotnet_root = Environment.GetEnvironmentVariable("DOTNET_ROOT");
+            if (!string.IsNullOrEmpty(dotnet_root))
+            {
+                // DOTNET_ROOT can be a path to dotnet OR a path to the folder containing dotnet.
```

Comment:
I think this can only be a folder.

https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables

Also:
https://github.com/dotnet/runtime/blob/main/docs/design/features/host-components.md
https://github.com/dotnet/runtime/blob/10d9e729f9ece24f89d73a5c6df18371c8db9360/src/installer/tests/HostActivation.Tests/MultiArchInstallLocation.cs#L119